src/Raven.Server/Documents/Indexes/Persistence/Lucene/OutputReduceIndexWriteOperation.cs

```diff
@@ -30,10 +30,10 @@ public OutputReduceIndexWriteOperation(MapReduceIndex index, LuceneVoronDirector
 
         public override void Commit(IndexingStatsScope stats)
         {
-            var enqueue = DocumentDatabase.TxMerger.Enqueue(_outputReduceToCollectionCommand);
-            base.Commit(stats);
             try
             {
+                var enqueue = DocumentDatabase.TxMerger.Enqueue(_outputReduceToCollectionCommand);
+                base.Commit(stats);
```

Comment:
Can you move base.Commit outside the try/catch? By doing this you will ensure that the exception message is appropriate.